src/Persistence/Mapping/AbstractClassMetadataFactory.php

```diff
@@ -462,7 +468,7 @@ public function isTransient($className)
         }
 
         // Check for namespace alias
-        if (strpos($className, ':') !== false) {
+        if (! class_exists($className) && strpos($className, ':') !== false) {
```

Comment:
I would use `! class_exists($className, false)`, to avoid triggering autoloaders for namespace aliases. Excluding anonymous classes does not need to trigger the autoloader.

Reply:
I think that makes sense, since we want to test if the class is anonymous and anonymous class probably don't require any autoloading.